components/org.wso2.carbon.identity.oauth.uma.common/src/main/java/org/wso2/carbon/identity/oauth/uma/common/JdbcUtils.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth.uma.common;
+
+import org.apache.axiom.om.OMElement;
+import org.wso2.carbon.database.utils.jdbc.NamedJdbcTemplate;
+import org.wso2.carbon.identity.base.IdentityRuntimeException;
+import org.wso2.carbon.identity.core.util.IdentityConfigParser;
+import org.wso2.carbon.identity.core.util.IdentityCoreConstants;
+
+import javax.naming.Context;
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+import javax.sql.DataSource;
+import javax.xml.namespace.QName;
+
+/**
+ * This class checks whether the database is created, if not it creates one. It reads
+ * the data source properties from the identity.xml.
+ */
+public class JdbcUtils {
```

Comment:
Don't we have utils in identity framework to find the value for this property. Check the OAuth DAOs to get an idea

Reply:
There was no method to return the data source at JDBCPersistenceManager. Therefore a method was added to return the data source https://github.com/wso2/carbon-identity-framework/pull/1617